### PR TITLE
7zip: fix legacysupport linking on macOS 10.12 and older

### PR DESCRIPTION
**Pull request by aeiouaeiouaeiouaeiouaeiouaeiou**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.8.5 12F2560 x86_64
Xcode 5.1.1 5B1008

GitHub Actions CI build

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: archivers/7zip/Portfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ legacysupport.newest_darwin_requires_legacy 16
 
 name                7zip
 version             22.00
-revision            0
+revision            1
```
Comment (mascguy) [inline archivers/7zip/Portfile:13]: If this is purely a fix for build failures on 10.13 and earlier, there's no need for a rev-bump.

File: archivers/7zip/Portfile [hunk 2]
```diff
@@ -51,6 +51,15 @@ compiler.blacklist-append  {*gcc-[3-4].*} {clang < 800} {macports-clang-3.*}
...
 
+# Force dynamic linking to use libMacportsLegacySupport.
+# Required only on unsupported systems.
+if {${os.platform} eq "darwin" && ${os.major} < 16} {
+    post-patch {
+        reinplace "s|LDFLAGS = $\(\LDFLAGS_STATIC\)\|LDFLAGS += $\(\LDFLAGS_STATIC\)\|" \
+        ${worksrcpath}/CPP/7zip/7zip_gcc.mak
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (aeiouaeiouaeiouaeiouaeiouaeiou) [reply archivers/7zip/Portfile:61]: @mascguy It should apply to all systems where legacysupport is used - I fixed that.

[the remark this answers is not in the document]
Comment (mascguy) [reply archivers/7zip/Portfile:61]: What about the build failures for 10.13? Or will that need to be resolved via a different method?

https://build.macports.org/builders/ports-10.13_x86_64-builder/builds/155153/steps/install-port/logs/stdio

CC: @kencu

[the remark this answers is not in the document]
Comment (aeiouaeiouaeiouaeiouaeiouaeiou) [reply archivers/7zip/Portfile:61]: We need to dig into the [AES detection code](https://github.com/kornelski/7z/blob/ec44a8a0700a8b2444b07f576be332f756754323/C/AesOpt.c), but first I want to fix the build on older macs.

[the remark this answers is not in the document]
Comment (mascguy) [reply archivers/7zip/Portfile:61]: Fair enough, we can tackle that separately then.

**Review by mascguy — APPROVED** [review]

**Pull request merged**